File: NewFlowar/NewFlowar/NewFlowar/Model/Cell.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;

namespace NewFlowar.Model
{
    public class Cell
    {
        public Map Map { get; set; }
        public float Height { get; set; }
        public Point Coord { get; set; }
        public Vector2 Location { get; set; }
        public Dictionary<int, Cell> Neighbourghs { get; set; }
        public Dictionary<int, int> Points { get; set; }
        public bool ContainsMinion { get; set; }

        public Cell(Map map, int x, int y, float left, float top)
        {
            this.Map = map;
            this.Coord = new Point(x, y);
            this.Location = new Vector2(left, top);

            this.Neighbourghs = new Dictionary<int, Cell>();
            this.Neighbourghs.Add(1,null);
            this.Neighbourghs.Add(2,null);
            this.Neighbourghs.Add(3,null);
            this.Neighbourghs.Add(4,null);
            this.Neighbourghs.Add(5,null);
            this.Neighbourghs.Add(6,null);

            this.Points = new Dictionary<int, int>();
            this.Points.Add(1, -1);
            this.Points.Add(2, -1);
            this.Points.Add(3, -1);
            this.Points.Add(4, -1);
            this.Points.Add(5, -1);
            this.Points.Add(6, -1);
        }

        public int IndexPosition
        {
            get
            {
                return (this.Coord.Y - 1) * this.Map.Width + this.Coord.X-1;
            }
        }

        public override string ToString()
        {
            return String.Format("{0} : {1},{2}", IndexPosition, Coord.X, Coord.Y);
        }
    }
}

File: PLZ/src/plz/model/liplz/Cell.java
package plz.model.liplz;

import plz.engine.Common;

import com.badlogic.gdx.math.Vector2;

public class Cell implements Cloneable
{
	public Map Map;
	public Vector2 Coord;
	public Vector2 Location;
	public Vector2 InitialLocation;
	public Cell[] Neighbourghs;
	public Tile Tile;

	public CellState State;

	public Cell()
	{
		
	}
	
	public Cell(Map map, int x, int y, float left, float top)
	{
		this.Map = map;
		this.Coord = new Vector2(x, y);
		this.Location = new Vector2(left, top);
		this.InitialLocation = new Vector2(left, top);

		this.Neighbourghs = new Cell[6];
		State = CellState.Visible;
		
		this.Tile = new Tile();
	}

	public int IndexPosition()
	{
		return (int) ((this.Coord.y - 1f) * this.Map.Width/2 + this.Coord.x - 1f);
	}

	public Cell GetDirection(int direction, int iteration)
	{
		Cell cell = this;
		for (int i = 0; i < iteration && cell != null; i++)
		{
			cell = cell.Neighbourghs[direction];
		}

		return cell;
	}

	@Override
	public Object clone()
	{
		Cell cell = new Cell();

		if (this.Coord != null)
			cell.Coord = new Vector2(this.Coord.x, this.Coord.y);
		if (this.InitialLocation != null)
			cell.InitialLocation = new Vector2(this.InitialLocation.x, this.InitialLocation.y);
		cell.Location = new Vector2(this.Location.x, this.Location.y);
		cell.Map = this.Map;
		
		return cell;
	}
}
